speechbrain/lobes/models/discrete/wavTokenzier.py

```diff
@@ -0,0 +1,157 @@
+"""This lobe enables the integration of  pretrained WavTokenizer.
+
+Note that you need topip install git+https://github.com/Tomiinek/WavTokenizer` to use this module.
+
+Repository: https://github.com/jishengpeng/WavTokenizer/
+Paper: https://arxiv.org/abs/2408.16532
+
+Authors
+ * Pooneh Mousavi 2024
+"""
+
+import os
+
+import torch
+import torch.nn as nn
+from huggingface_hub import snapshot_download
+
+
+class WavTokenizer(nn.Module):
+    """An wrapper for the WavTokenizer model
+
+    Arguments
+    ---------
+    source : str
+        A HuggingFace repository identifier or a path
+    save_path : str
+        The location where the pretrained model will be saved
+    config : str
+        The name of the config file.
+    checkpoint : str
+        The name of the checkpoint file.
+    sample_rate : int (default: 24000)
+        The audio sampling rate
+    freeze : bool
+        whether the model will be frozen (e.g. not trainable if used
+        as part of training another model)
+
+    Example
+    -------
+    >>> model_hub = "novateur/WavTokenizer"
+    >>> save_path = "savedir"
+    >>> config="wavtokenizer_smalldata_frame40_3s_nq1_code4096_dim512_kmeans200_attn.yaml"
+    >>> checkpoint="WavTokenizer_small_600_24k_4096.ckpt"
+    >>> model = WavTokenizer(model_hub, save_path,config=config,checkpoint=checkpoint)
+    >>> audio = torch.randn(4, 48000)
+    >>> length = torch.tensor([1.0, .5, .75, 1.0])
+    >>> tokens, embs= model.encode(audio)
+    >>> tokens.shape
+    torch.Size([4, 1, 80])
+    >>> embs.shape
+    torch.Size([4, 80, 512])
+    >>> rec = model.decode(tokens)
+    >>> rec.shape
+    torch.Size([4, 48000])
+    """
+
+    def __init__(
+        self,
+        source,
+        save_path=None,
+        config="wavtokenizer_smalldata_frame40_3s_nq1_code4096_dim512_kmeans200_attn.yaml",
+        checkpoint="WavTokenizer_small_600_24k_4096.ckpt",
+        sample_rate=24000,
+        freeze=True,
+    ):
+        # Lazy import to avoid circular dependency issues
+        try:
+            import wavtokenizer
+
+            self.wavtokenizer = wavtokenizer
+        except ImportError:
+            raise ImportError(
+                "Please install the WavTokenizer module using: "
+                "`pip install git+https://github.com/Tomiinek/WavTokenizer`"
+            )
+
+        super().__init__()
+
+        path = snapshot_download(repo_id=source, cache_dir=save_path)
+        checkpoint_path = os.path.join(path, checkpoint)
+        config_path = os.path.join(path, config)
+        self.model = self.wavtokenizer.WavTokenizer.from_pretrained0802(
```

Comment:
Please note that the functions for loading the model are specific to their implementation and are not part of the Hugging Face Transformers library.

Reply:
Can't we rely on the official one? Checking what they do that is so special to re do it afterwards. Asking this because relying on obscure external function implies that we will 100% have to fix bugs in the near future because they changed their API or because it does not work anymore with the format of the file in the HF hub etc etc. Like it will 100000% happen. If the code is in speechbrain, we can maintain it.